pandas/core/arrays/datetimes.py

```diff
@@ -399,9 +404,12 @@ def dtype(self):
             If the values are tz-aware, then the ``DatetimeTZDtype``
             is returned.
         """
-        if self.tz is None:
-            return _NS_DTYPE
-        return DatetimeTZDtype('ns', self.tz)
+        return self._dtype
+
+    @property
```

Comment:
Hmm why was this necessary? Ideally everyone would reference `self.dtype.tz`

Reply:
I think if you update the `tz` property defend below to be 

```python
    @property
    def tz(self):
        """
        Return timezone, if any.

        Returns
        -------
        datetime.tzinfo, pytz.tzinfo.BaseTZInfo, dateutil.tz.tz.tzfile, or None
            Returns None when the array is tz-naive.
        """
        # GH 18595
        return getattr(self.dtype, 'tz', None)

    @tz.setter
    def tz(self, value):
        # GH 3746: Prevent localizing or converting the index by setting tz
        raise AttributeError("Cannot directly set timezone. Use tz_localize() "
                             "or tz_convert() as appropriate")
```

then things will hopefully work.